dora/core/common/src/main/java/alluxio/membership/WorkerServiceEntity.java

```diff
@@ -108,7 +108,8 @@ public boolean equals(Object o) {
     // only the service name and the identity are the factors that define
     // what a WorkerServiceEntity is. Any other is either ephemeral or supplementary data.
     return mIdentity.equals(anotherO.mIdentity)
-        && getServiceEntityName().equals(anotherO.getServiceEntityName());
+        && getServiceEntityName().equals(anotherO.getServiceEntityName())
+        && mAddress.equals(anotherO.getWorkerNetAddress());
```

Comment:
The comment explicitly states that `WorkerServiceEntity` ignores worker address for equality comparison. If a worker with the same ID restarts and the address changes, it will leave two different entries on etcd and causes the clients to think there are multiple workers, but there are actually only one worker.

Reply:
Bowen, actually when collision happens, in etcd comparison it directly compares the value bytes (EtcdMembershipManager.java line 134) and under that situation we evaluate if there's a need to update, and then we use equals here, if equals we know its the same worker and then update with new info, this is to introduce backward compatibility after adding a field "httpport" into the info to persist on etcd. Can u check if this equals is used somewhere else apart from etcdmembership join that will cause other problems?